drivers/include/mcp47xx.h

```diff
@@ -0,0 +1,347 @@
+
+    mcp47xx_variant_t   variant;    /**< used variant of MCP47xx */
+    mcp47xx_gain_t      gain;       /**< Gain selection */
+    mcp47xx_vref_t      vref;       /**< Voltage reference selection */
+    mcp47xx_pd_mode_t   pd_mode;    /**< Power-down mode selection */
+
+} mcp47xx_params_t;
+
+/**
+ * @brief   MCP47xx device data structure type
+ */
+typedef struct {
+    mcp47xx_params_t params;              /**< device configuration parameters */
+    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set */
+} mcp47xx_t;
+
+#if IS_USED(MODULE_SAUL) || DOXYGEN
+/**
+ * @brief   MCP47xx configuration structure for mapping DAC channels to SAUL
+ */
+typedef struct {
+    const char *name;   /**< name of the MCP47xx device */
+    unsigned int dev;   /**< index of the MCP47xx device */
+    dac_t channel;      /**< channel of the MCP47xx device */
+    uint16_t initial;   /**< initial value */
+} mcp47xx_saul_dac_params_t;
+#endif
+
+/**
+ * @brief   Initialize the MCP47xx DAC
+ *
+ * All expander pins are set to be input and are pulled up.
+ *
+ * @param[in]   dev     descriptor of the MCP47xx DAC device
+ * @param[in]   params  configuration parameters, see #mcp47xx_params_t
+ *
+ * @retval  MCP47XX_OK      on success
+ * @retval  MCP47XX_ERROR_* a negative error code on error,
+ *                           see #mcp47xx_error_codes_t
+ */
+int mcp47xx_init(mcp47xx_t *dev, const mcp47xx_params_t *params);
+
+/**
+ * @brief   Initialize a MCP47xx DAC channel
+ *
+ * After the initialization, the DAC channel is active and its output is set
+ * to 0.
+ *
+ * @param[in]   dev     descriptor of the MCP47xx DAC device
+ * @param[in]   chn     channel to initialize
+ *
+ * @retval  MCP47XX_OK          on success
+ * @retval  MCP47XX_ERROR_*     a negative error code on error,
+ *                              see #mcp47xx_error_codes_t
+ */
+int mcp47xx_dac_init(mcp47xx_t *dev, dac_t chn);
+
+/**
+ * @brief   Write a value to a MCP47xx DAC channel
+ *
+ * The value is always given as 16-bit value and is internally scaled to the
+ * actual resolution that the DAC unit provides, e.g., 12-bit. So to get the
+ * maximum output voltage, this function has to be called with value set
+ * to 65535 (UINT16_MAX).
+ *
+ * @param[in]   dev     descriptor of the MCP47xx DAC device
+ * @param[in]   chn     channel to set
+ * @param[in]   value   value to set line to
+ *
+ * @retval  none
+ */
+void mcp47xx_dac_set(mcp47xx_t *dev, dac_t chn, uint16_t value);
+
+/**
+ * @brief   Get the current value of a MCP47xx DAC channel
+ *
+ * The value is always given as 16-bit value and is internally scaled to the
+ * actual resolution that the DAC unit provides, e.g., 12-bit.
+ *
+ * @param[in]   dev     descriptor of the MCP47xx DAC device
+ * @param[in]   chn     channel to set
+ * @param[out]  value   value to set line to
+ *
+ * @retval  none
+ */
+void mcp47xx_dac_get(mcp47xx_t *dev, dac_t chn, uint16_t *value);
```

Comment:
I'm a bit surprised you implemented this when `periph/dac` API does not provide any `dac_get()`.
Feel free to keep it though :)